src/checkpoint.rs

```diff
@@ -11,5 +11,7 @@ pub fn checkpoint<Load, Resolve, Instruction, Resolution>
           Instruction: Serialize + Deserialize + Eq,
           Resolution: Serialize + Deserialize
 {
+    ::std::mem::drop(load);
+    ::std::mem::drop(resolve);
```

Comment:
Clippy complains that `load` and `resolve` are consumed by the function, and suggests they be turned into references. I didn't want to change the signature, so I shut up clippy by explicitly using them. (Although it turns out this module is completely unused, and I have a pending branch deleting a bunch of unused code, including this module).